src/services/__tests__/LengthAwarePaginator.spec.js

```diff
@@ -0,0 +1,190 @@
+import LengthAwarePaginator from '../pagination/LengthAwarePaginator';
+import {
+  mockData,
+  mockDataFirstItem,
+  mockDataLastItem,
+} from '../../utils/__mocks__/db';
+import LesgoException from '../../exceptions/LesgoException';
+import db from '../../utils/db';
+
+jest.mock('../../utils/db');
+
+beforeEach(() => {
+  db.select.mockClear();
+});
+
+const FILE = 'Services/pagination/LengthAwarePaginator';
+
+describe('test LengthAwarePaginator instantiate', () => {
+  it('should not throw exception when instantiating', async () => {
+    const paginator = new LengthAwarePaginator(
+      db,
+      'SELECT * FROM tests',
```

Comment:
I didn't notice that we should pass a query statement inside the length aware paginator, how about an interface of collection to be passed, where we have another class that decorates a `handle()` or `collection()`

So this class will never be dependent to call a real database, we can just create a class that returns a collection (easy way of mocking it)

Just a suggestion that we can improve soon.

Reply:
plus one to this. clean code

Reply:
This is a simple paginator for now, to allow our current AuroraDbService to integrate with this Paginator class. We'll definitely find better ways to achieve this in later iterations